### PR TITLE
Search date format

### PR DESCRIPTION
**Pull request by will-moore**
This silently ignores any invalid date formats in the search form. See https://trello.com/c/ai1te7XA/40-search-date-formatting

 - To test, go to search page and click on Start Date input.
 - The date-picker will popup but it is still possible to type numbers into the input.
 - Type in an invalid date (random numbers) and perform search.
 - Invalid date should be ignored (not throw exception).

### CODE REVIEW

**Comment by gusferguson** [comment]
@will-moore 

Tested using cowfish - as per standup link - user-3 read-only-1

Behaves as expected.
Good to merge.

**Comment by joshmoore** [comment]
@will-moore : looking at this it makes sense, but I know I'm not the best with timestamps. Would unit tests of the parsing be possible? I.e. what do you expect to be ignored.

**Comment by will-moore** [comment]
@joshmoore see previous commit "Date parsing moved to utils and unit test added"

**Comment by gusferguson** [comment]
@will-moore 

Tested using https://cowfish.openmicroscopy.org/merge/webclient/l user-3 read-only-1

Behaves as expected.
Good to merge.

**Comment by jburel** [comment]
@gusferguson: could you check that both client have the same behaviour in case we need to adjust anything in insight?

**Comment by gusferguson** [comment]
@jburel - OK

**Comment by gusferguson** [comment]
@jburel 

Tested with OMERO.insight-5.2.1-271-126effd-ice35-b210-win eel user-3 read-only-1

Insight behaves as expected - does not recognise invalid numbers in either date field and returns unchanged results.

File: components/tools/OmeroWeb/omeroweb/webclient/controller/search.py [hunk 1]
```diff
@@ -68,20 +68,24 @@ def search(self, query, onlyTypes, fields, searchGroup, ownedBy,
...
-                                      d2.microsecond) * 1000)]
+                try:
+                    d1 = getDateTime(p[0]+" 00:00:00")
+                    d2 = getDateTime(p[1]+" 23:59:59")
+
+                    created = [rtime(long(time.mktime(d1.timetuple()) + 1e-6 *
+                                          d1.microsecond) * 1000),
+                               rtime(long(time.mktime(d2.timetuple()) + 1e-6 *
+                                          d2.microsecond) * 1000)]
+                    print 'created', created
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroWeb/omeroweb/webclient/controller/search.py]: Print?

File: components/tools/OmeroWeb/omeroweb/webclient/controller/search.py [hunk 2]
```diff
@@ -68,20 +68,24 @@ def search(self, query, onlyTypes, fields, searchGroup, ownedBy,
         if len(onlyTypes) == 1:
             batchSize = 1000
         if date is not None:
+            print 'date', date
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroWeb/omeroweb/webclient/controller/search.py]: Print?

**Comment by will-moore** [comment]
Removed print statements.

**Comment by jburel** [comment]
@bramalingam: do you have a robot test for that?

**Comment by bramalingam** [comment]
I do check the date field with a valid format and search, but don't have a false positive check as suggested above.

**Comment by gusferguson** [comment]
@will-moore 

Retested with https://cowfish.openmicroscopy.org/merge/webclient/ user-3 read-only-1

Behaves as expected.
Good to merge.

**Comment by jburel** [comment]
Robot test can be added in another PR. Merging

**Pull request merged**